src/economy/ship_fleet.cc

```diff
@@ -723,232 +675,30 @@ void ShipFleet::act(Game& game, uint32_t /* data */) {
 		return;
 	}
 
-	if (!active()) {
-		// If we are here, most likely act() was called by a port with waiting wares or
-		// with an expedition ready, although there are still no ships.
-		// We can't handle it now, so we reschedule the act()
-		schedule_act(game, kFleetInterval);  // retry in the next time
-		act_pending_ = true;
-		return;
-	}
-
 	molog("ShipFleet::act\n");
 
-	// For each waiting port, try to find idle ships and send to it the closest one.
-	uint16_t waiting_ports = ports_.size();
-	for (PortDock* p : ports_) {
-		if (p->get_need_ship() == 0) {
-			--waiting_ports;
-			continue;
-		}
-
-		Ship* closest_ship = nullptr;
-		uint32_t shortest_dist = kRouteNotCalculated;
-		bool waiting = true;
-
-		for (Ship* s : ships_) {
-			bool fallback = false;
-			if (s->get_current_destination(game)) {
-				if (s->has_destination(game, *p)) {
-					waiting = false;
-					--waiting_ports;
-					break;
-				}
-				fallback = true;  // The ship already has a destination
-			}
-			if (s->get_ship_state() != Ship::ShipStates::kTransport) {
-				continue;  // Ship is not available, e.g. in expedition
-			}
-
-			// Here we get distance ship->port
-			uint32_t route_length = kRouteNotCalculated;
-
-			// Get precalculated distance for ships available at ports
-			{
-				PortDock* cur_port = get_dock(game, s->get_position());
-				if (cur_port) {          // Ship is at a port
-					if (cur_port == p) {  // Same port
-						route_length = 0;
-					} else {  // Different port
-						Path precalculated_path;
-						if (get_path(*cur_port, *p, precalculated_path)) {
-							route_length = precalculated_path.get_nsteps();
-						}
-					}
-				}
-			}
-
-			// Get distance for ships available but not at a port (should not happen frequently)
-			if (route_length == kRouteNotCalculated) {
-				route_length = s->calculate_sea_route(game, *p);
-			}
-			if (fallback) {
-				if (!penalize_route(game, *p, *s, &route_length)) {
-					continue;
-				}
-			}
-
-			if (route_length < shortest_dist) {
-				shortest_dist = route_length;
-				closest_ship = s;
-			}
-		}
+	log("\n\n--- NOCOM ---\n");
+	log_general_info(game);
```

Comment:
I assume we remove this before merge

Reply:
Yep, here and in shipping_schedule.cc are two NOCOMs which are very useful for debugging but will have to go just before merging.